frontend/src/components/Credentials/GSecretDialogGeneric.vue

```diff
@@ -116,39 +109,41 @@ export default {
         this.$emit('update:modelValue', modelValue)
       },
     },
+    data: {
+      get () {
+        return this.internalYaml
+      },
+      set (value) {
+        this.internalYaml = value
+        this.touched = true
+        this.stringData = {}
+
+        try {
+          this.stringData = yaml.load(value)
+        } catch (err) {
+        /* ignore errors */
+        } finally {
+          if (!isObject(this.stringData)) {
+            this.stringData = {}
+          }
+        }
+      },
+    },
     valid () {
       return !this.v$.$invalid
     },
     isCreateMode () {
       return !this.secret
     },
   },
-  methods: {
-    onInputSecretData () {
-      this.secretData = {}
-
-      try {
-        this.secretData = yaml.load(this.data)
-      } catch (err) {
-        /* ignore errors */
-      } finally {
-        if (!isObject(this.secretData)) {
-          this.secretData = {}
-        }
-      }
-
-      this.v$.data.$touch()
-    },
-    reset () {
-      this.v$.$reset()
-
-      this.data = ''
-      this.secretData = {}
-
-      if (!this.isCreateMode) {
-        setDelayedInputFocus(this, 'data')
+  watch: {
+    stringData (value) {
+      if (!this.touched) {
+        this.internalYaml = yaml.dump(value)
```

Comment:
adding a secret with 
```yaml
foo: ~
```
and pressing save will result in
```yaml
foo: 'null'
```

Reply:
fixed